src/driver/amdxdna/aie2_ctx.c

```diff
@@ -332,12 +334,19 @@ aie2_sched_job_run(struct drm_sched_job *sched_job)
 		goto out;
 	}
 
-	if (amdxdna_cmd_get_op(cmd_abo) == ERT_CMD_CHAIN)
-		ret = aie2_cmdlist_multi_execbuf(ctx, job, aie2_sched_cmdlist_resp_handler);
+	/*
+	 * Transaction binaries are only supported with RAI 1.5 release onwards. Below
+	 * implementation returns -EOPNOTSUPP error code for any older firmware versions.
+	 */
+	class = aie2_is_supported_msg(ctx->client->xdna->dev_handle, MSG_OP_CMD_CHAIN_NPU) ?
```

Comment:
The class is obtained from 'ctx'. Maybe we can save it in ctx when it is created so that we don't have to pass it around